bin/i18n/sync-out.rb

```diff
@@ -258,6 +258,22 @@ def sanitize_data_and_write(data, dest_path)
   File.write(dest_path, dest_data)
 end
 
+def sort_and_sanitize(hash)
+  hash.sort_by {|key, _| key}.each_with_object({}) do |(key, value), result|
+    case value
+    when Hash
+      sorted_value = sort_and_sanitize(value)
+      result[key] = sorted_value unless sorted_value.nil? || sorted_value.empty?
```

Comment:
1. It can't be nil as `when Hash` allows only hash values
2. There is no need to sort an empty hash value
```suggestion
      result[key] = sort_and_sanitize(value) unless sorted_value.empty?
```